clients/src/main/java/org/apache/kafka/clients/admin/ApiVersionRange.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.admin;
+
+import org.apache.kafka.common.errors.UnsupportedVersionException;
+
+import java.util.Objects;
+
+/**
+ * A public class which represents a range of versions.
+ */
+public class ApiVersionRange {
+    /**
+     * The ID of the API.
+     */
+    private final int id;
+
+    /**
+     * The lowest supported version.
+     */
+    private final int lowest;
+
+    /**
+     * The highest supported version.
+     */
+    private final int highest;
+
+    public ApiVersionRange(int id, int lowest, int highest) {
```

Comment:
Should we include a name for the API? We would have to make it clear that the name is like `toString` in a class, it may change, but it helps understand what one is talking about.